### PR TITLE
Limit version of flit_core specified to build with

### PR DESCRIPTION
**Pull request by takluyver**
I suggest every package built with Flit specifies a maximum version, so old packages can continue to be built in the future, even if things change in newer versions of Flit.

The recommendations around this are here: https://flit.pypa.io/en/stable/pyproject_toml.html

As a concrete example, the Flit-specific `[tool.flit.metadata]` table is gradually being replaced by the standardised `[project]` table. Flit 3.x has supported both options, but my plan for v4 is to only support `[project]`.

### CODE REVIEW

**Comment by takluyver** [comment]
The CI failure is not related to this change - I guess `black` has changed a bit of it's style for 2024.

File: pyproject.toml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [build-system]
-requires = ["flit_core"]
+requires = ["flit_core >=2,<4"]
 build-backend = "flit_core.buildapi"
 
 [tool.flit.metadata]
```
Comment (ogrisel) [inline pyproject.toml:5]:
```suggestion
# TODO: replace the following section by the standard [project] section
# to be able to use flit v4.
[tool.flit.metadata]
```

**Review by ogrisel — APPROVED** [review]
Thanks for the PR. LGTM with the following TODO:

**Review by jeremiedbb — APPROVED** [review]
Thanks @takluyver

**Pull request merged**